src/Power.cpp

```diff
@@ -379,9 +385,15 @@ class AnalogBatteryLevel : public HasBatteryLevel
 #endif
 #ifdef EXT_CHRG_DETECT
         return digitalRead(EXT_CHRG_DETECT) == ext_chrg_detect_value;
-#else
-        return isBatteryConnect() && isVbusIn();
+
+#elif HAS_TELEMETRY && !defined(ARCH_PORTDUINO) && !defined(HAS_PMU) && !MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR
+        if (hasINA()) {
+            return (getINACurrent() > 0) ? OptTrue : OptFalse;
```

Comment:
One more thought actually. I've never used these current sensors, but is there a chance that some users have wired these with the current flowing the other way?

Reply:
> One more thought actually. I've never used these current sensors, but is there a chance that some users have wired these with the current flowing the other way?

I have these sensors wired to measure power usage by the device itself (rather than net current to/from the battery), so this value will always be greater than zero and so this new behaviour would be surprising...

However I also use the three channel INA3221 to measure the device voltage/current on one channel, and solar input on another channel. I've considered using the third channel to measure net flow to the battery, which I think is the assumption being made here for the expected wiring...

Perhaps a configuration item for "INA address and channel" for each of "device" and "battery" power makes sense? Or some configuration for the criteria used to determine charging/not charging... :shrug:

Reply:
But the direct answer to your question is probably yes, I'm sure these get wired "wrong", and if you are only measuring current in one direction, you could just ignore the sign.

Reply:
this is incorrect; they don't care about direction -- I have wired mine such that negative current means the battery is discharging and positive current means it is charging. I spoke with @todd-herbert in Discord and will make a note of this in the source.

Reply:
I don't completely follow, but this behaviour is definitely peculiar to a specific configuration which I don't believe to be a given...

Reply:
> I have these sensors wired to measure power usage by the device itself (rather than net current to/from the battery), so this value will always be greater than zero and so this new behaviour would be surprising...

This is the scenario I could imagine happening.

> But the direct answer to your question is probably yes, I'm sure these get wired "wrong", and if you are only measuring current in one direction, you could just ignore the sign.

Also the device would report as "powered" when discharging, and "on battery" when charging, I think?